client/os.cpp

```diff
@@ -397,6 +404,39 @@ namespace tec {
 	}
 
 	void OS::DispatchMouseMoveEvent(const double x, const double y) {
+		if (OS::mouse_locked) {
+			// mouse lock is where we hide the cursor and constrain it to the window
+			// we also request raw mouse motion if available
+			std::shared_ptr<MouseMoveEvent> mmov_event = std::make_shared<MouseMoveEvent>(
+				MouseMoveEvent{
+					x / this->client_width,
+					y / this->client_height,
+					static_cast<int>(this->old_mouse_x),
+					static_cast<int>(this->old_mouse_y),
+					static_cast<int>(x),
+					static_cast<int>(y)
+				});
+			EventSystem<MouseMoveEvent>::Get()->Emit(mmov_event);
+			double client_center_x = static_cast<double>(this->client_width / 2);
+			double client_center_y = static_cast<double>(this->client_height / 2);
+			// constrain the mouse towards the center of the window
+			// we do this to prevent an initial frame of "overshoot"
+			// and also to prevent anything using absolute mode from wandering away, i.e. tooltip window
+			if (x < client_center_x && x > this->old_mouse_x) {
```

Comment:
This could a great help function `bool IsBetween(T val, T lhs, T rhs)` and I am sure it could be useful in other locations.